extensions/extension.go

```diff
@@ -137,6 +138,28 @@ func initInternal(c *Config) error { //nolint:funlen,gocognit,gocyclo,maintidx /
 			return err
 		}
 	}
+	// rand() and rand(n) functions.
+	randFn := object.Extension{
+		Name:     "rand",
+		MinArgs:  0,
+		MaxArgs:  1,
+		ArgTypes: []object.Type{object.INTEGER},
+		Callback: func(env any, _ string, args []object.Object) object.Object {
+			eval.TriggerNoCache(env)
+			if len(args) == 0 {
+				return object.Float{Value: rand.Float64()} //nolint:gosec // no need for crypto/rand here.
+			}
+			n := args[0].(object.Integer).Value
+			if n <= 0 {
+				return object.Error{Value: "argument to rand() if given must be > 0, >=2 for something useful"}
```

Comment:
I don't get it. It's because it's a variadic right?

The error message is unclear

I mean if n<=0,means you expect arguments,

But then the message talk about >=2

Why not checking >=2 then?

Reply:
it's not technically an error to do rand(1) but it will always produce 0, yet maybe some code needs a random based on some other value that can change so, like the underlying go function it doesn't fail but if someone does make the mistake of calling with 0 they may read the message and realize they should probably have 2 as the minimum, could be a warning but there isn't a grol side warning